### PR TITLE
OpenAI Compatible API TTS Provider

### LINKED ISSUE #2639: [FEATURE_REQUEST] TTS: OpenAI-compatible custom server support (e.g. openedai-speech)
### Have you searched for similar requests?

Yes

### Is your feature request related to a problem? If so, please describe.

The OpenAI option for the TTS extension only supports the official OpenAI server. Custom OpenAI-compatible servers can't be specified.

### Describe the solution you'd like

It would be great to be able to enter a custom URL for the OpenAI TTS support so that alternatives like [openedai-speech](https://github.com/matatonic/openedai-speech) can be used. This is supported by Open WebUI, so it would allow this one solution to work with both projects.

### Describe alternatives you've considered

Using whatever the (custom) Chat Generation endpoint is set as would *almost* work, but openedai-speech doesn't implement chat completion (not its goal), so a middle-server would be needed to redirect requests appropriately, which isn't great.

### Additional context

_No response_

### Priority

Low (Nice-to-have)

### Are you willing to test this on staging/unstable branch if this is implemented?

Yes

### PR DESCRIPTION
**Pull request by ayancey**
## Checklist:

- [X] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

This closes #2639.

### CODE REVIEW

**Comment by ayancey** [comment]
You might be wondering, why not modify the original OpenAI provider?

Unlike OpenAI, the provider may have custom voices, but there's no endpoint to retrieve them. Because of that, the user needs to manually input the list of voices. There's no URLs for the preview sounds, so we need to generate them on the fly the same way that speecht5 does it.

**Comment by Cohee1207** [comment]
> Unlike OpenAI, the provider may have custom voices, but there's no endpoint to retrieve them. Because of that, the user needs to manually input the list of voices. There's no URLs for the preview sounds, so we need to generate them on the fly the same way that speecht5 does it.

I tried modifying the original provider but abandoned that idea for this exact reason. So I'm not opposed to have it being separate.

File: public/scripts/extensions/tts/openai-compatible.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        this.audioElement.src = url;
+        this.audioElement.play();
+        this.audioElement.onended = () => URL.revokeObjectURL(url);
+    }
+
+    async fetchTtsGeneration(inputText, voiceId) {
+        console.info(`Generating new TTS for voice_id ${voiceId}`);
+        const response = await fetch(this.settings.provider_endpoint, {
+            method: 'POST',
+            headers: getRequestHeaders(),
```
Comment (Cohee1207) [inline public/scripts/extensions/tts/openai-compatible.js:175]: Any possibility to also add support for bearer authentication header?

Comment (ayancey) [reply public/scripts/extensions/tts/openai-compatible.js:137]: can you discord dm me? username is ayancey

**Review by Cohee1207 — APPROVED** [review]
LGTM. Added "Remove key" to all instances of extensions using secrets

**Pull request merged**